src/stomp_moveit_planning_context.cpp

```diff
@@ -109,10 +109,13 @@ bool StompPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   const auto task = createStompTask(config, *this);
   stomp_ = std::make_shared<stomp::Stomp>(config, task);
 
-  // Timeout async task
+  std::condition_variable cv;
+  std::mutex cv_mutex;
+  bool finished;
   auto timeout_future = std::async(std::launch::async, [&, stomp = stomp_]() {
-    std::this_thread::sleep_for(std::chrono::duration<double>(req.allowed_planning_time));
-    if (stomp)
+    std::unique_lock<std::mutex> lock(cv_mutex);
+    cv.wait_for(lock, std::chrono::duration<double>(req.allowed_planning_time), [&finished]{return finished;});
+    if (stomp_ == stomp)
```

Comment:
```suggestion
    if (stomp && !finished)
```
What do you think about this version? We only care about the validity of the local (temporary) instance of stomp, and we only want to cancel an unfinished plan. I just tested this locally and it works just the same.